ui/pages/forgot-domain/index.js

```diff
@@ -0,0 +1,100 @@
+import { useState } from 'react'
+
+import LoginLayout from '../../components/layouts/login'
+
+export default function ForgotDomain() {
+  const [email, setEmail] = useState('')
+  const [error, setError] = useState('')
+  const [submitted, setSubmitted] = useState(false)
+
+  async function onSubmit(e) {
+    e.preventDefault()
+    setSubmitted(true)
+
+    try {
+      const res = await fetch('/api/forgot-domain-request', {
+        method: 'post',
+        body: JSON.stringify({
+          email,
+        }),
+      })
+
+      if (!res.ok) {
+        throw await res.json()
```

Comment:
I think throwing this here may cause the UI to crash. Better to handle it

Reply:
I patterned this off of how it's working in `signup/index.js`, `password-reset/index.js`, and `login/index.js` so I would assume it's broken in those places too? It looks like we're also doing it in `login/finish.js`. Is there a pattern I can follow to do it correctly?